### ISSUE TITLE
Wallet creation fails on C65 build

### ISSUE THREAD

**Issue by srirambv** [brave/browser-laptop#13366]
### Description
Wallet creation fails on C65 build

### Steps to Reproduce
  1. Clean install 0.21.654
  2. Launch from console and enable payments
  3. Gets stuck in wallet creation, reproduced by @bsclifton as well

**Actual result:**
Wallet creation fails with the following error on console
```
properties error: Error: Ledger client initialization incomplete.
(node:3896) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 removed listeners added. Use emitter.setMaxListeners() to increase limit
MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 removed listeners added. Use emitter.setMaxListeners() to increase limit
    at _addListener (events.js:262:19)
    at DownloadItem.addListener (events.js:279:10)
    at DownloadItem.<anonymous> (C:\Users\SriramBV\AppData\Local\BraveBeta\app-0.21.654\resources\app.asar\app\filtering.js:602:12)
    at emitTwo (events.js:106:13)
    at DownloadItem.emit (events.js:194:7)
properties error: Error: Ledger client initialization incomplete.
<<< GET https://mercury-proxy.privateinternetaccess.com/v3/publisher/identity?publisher=google.com
<<< content-type: application/json; charset=utf-8
<<< user-agent: Brave/0.21.654 Chrome/65.0.3325.88 Muon/5.1.3 Microsoft Windows x64
<<< accept-encoding:
<<<
[3896:1228:0302/105054.807:ERROR:platform_sensor_reader_win.cc(239)] NOT IMPLEMENTED
[ response for GET https://mercury-proxy.privateinternetaccess.com/v3/publisher/identity?publisher=google.com ]
```

**Expected result:**
Wallet should get created when payments is enabled

**Reproduces how often:**
100% on 0.21.654 build

### Brave Version

**about:brave info:**

Brave | 0.21.654
-- | --
OS Release | 10.0.16299
Muon | 5.1.3
V8 | 6.5.254.26
libchromiumcontent | 65.0.3325.88
Node.js | 7.9.0
rev | bc6272c
Update Channel | Beta
OS Platform | Microsoft Windows
Brave Sync | v1.4.2
OS Architecture | x64

**Reproducible on current live release:**
No

**Comment by NejcZdovc** [comment]
https://github.com/brave-intl/bat-client/blob/573a72ed13aa6b60a8994d848a0f67919d18113e/index.js#L1358-L1390

this block is problematic `worker.onmessage` is not triggered, where `worker.postMessage` is triggered

**Comment by bsclifton** [comment]
@darkdh can you help check this out?

**Comment by hferreiro** [comment]
Following the `worker.postMessage` call, it looks like the `onmessage` callback isn't working on the muon side. In `OnMessageInternal` (https://github.com/brave/muon/blob/master/brave/common/workers/worker_bindings.cc#L34), the call to `onmessage_fun` (https://github.com/brave/muon/blob/master/brave/common/workers/worker_bindings.cc#L54) is skipped because the condition `onmessage->IsFunction()` is false.

**Comment by darkdh** [comment]
right, and you can assign other properties like string in JS and read it in C++ but not function

**Comment by darkdh** [comment]
The root cause is `env()->module_system()->Require(module_name_)` failure
https://github.com/brave/muon/blob/C65/brave/common/workers/v8_worker_thread.cc#L130

**Comment by darkdh** [comment]
real root cause is this commit
https://chromium.googlesource.com/chromium/src.git/+/312031eec9c97d5cb9dea801f4f3cefe2f3ad937%5E%21/

they removed
```
-      // AMD.
-      GetPropertyUnsafe(v8_context, define_object, "define"),
```

**Comment by bsclifton** [comment]
Fixed by upgrading Muon with 39573e464297e4a4583fbf7f665d3e463e9c5427
